### ISSUE TITLE
RangeError (index): Index out of range: index should be less than [WEB]

### ISSUE THREAD

**Issue by bkr32** [DavBfr/dart_pdf#459]
**Describe the bug**
Instead of Displaying the PDF, it returns a blank canvas with the error `RangeError (index): Index out of range: index should be less than 24674: 24674` 

**To Reproduce**
Code snippet to reproduce the behavior:
```
Container(
                              height: 900,
                              width: 600,
                              child: PdfPreview(
                                initialPageFormat: PdfPageFormat.a4,
                                maxPageWidth: 700.0,
                                canChangePageFormat: false,
                                actions: actions,
                                build: (format) =>
                                    reportView(format, name),
                              ))
  Future<Uint8List> reportView(PdfPageFormat format, name) async {
    final pdf = pw.Document();
    pdf.addPage(
      pw.Page(
        pageFormat: PdfPageFormat.a4,
        build: (context) {
          return pw.Center(
            child: pw.Text('name'),
          );
        },
      ),
    );

    return pdf.save();
  }
```

**Expected behavior**
Should Display the document normally

**Screenshots**

![image](https://user-images.githubusercontent.com/7794407/97111868-3874af80-16f2-11eb-819b-91ef3cb09c82.png)

**Flutter Doctor**

```
[√] Flutter (Channel beta, 1.23.0-18.1.pre, on Microsoft Windows [Version
    10.0.18363.1139])
    • Flutter version 1.23.0-18.1.pre at D:\flutter
    • Framework revision 198df796aa (10 days ago), 2020-10-15 12:04:33 -0700
    • Engine revision 1d12d82d9c
    • Dart version 2.11.0 (build 2.11.0-213.1.beta)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\32\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • CHROME_EXECUTABLE = C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.49.2)
    • VS Code at C:\Users\32\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.1

[√] VS Code, 64-bit edition (version 1.38.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.14.1

[√] Connected device (2 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 86.0.4240.75
```

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [* ] Browser

**Smartphone (please complete the following information):**
Device: [Desktop]
OS: [Windows 10]
Browser [Chrome]
Version [86.0.4240.75]

**Comment by DavBfr** [comment]
Can you try with async/await?

```dart
...
build: (format) async => await reportView(format, name),
...
```

**Comment by bkr32** [comment]
Still the same error :(

**Comment by justkawal** [comment]
@bkr32 I am also getting the same error!!

Did you find any work-around ?

**Comment by bkr32** [comment]
@justkawal nope, i ended up straight up creating the pdf and sharing/downloading it, there are a couple of other packages you can try out from pub.dev

**Comment by justkawal** [comment]
@DavBfr Are you gonna do something about it ?

**Comment by DavBfr** [comment]
@justkawal Yes, I see what's wrong. Maybe a breaking change in the latest Flutter dev channel.

**Comment by justkawal** [comment]
@DavBfr Thanks, Looking forward for for sooner resolution.

**Comment by DavBfr** [comment]
Can you test using the master branch? Adding this in your `pubspec.yaml` as a top-level element:

```yaml
dependency_overrides:
  printing:
    git:
      url: https://github.com/DavBfr/dart_pdf
      path: printing
```

**Comment by jrheisler** [comment]
> Can you test using the master branch? Adding this in your `pubspec.yaml` as a top-level element:
> 
> ```yaml
> dependency_overrides:
>   printing:
>     git:
>       url: https://github.com/DavBfr/dart_pdf
>       path: printing
> ```

That seems to fix the problem on my web app.

**Comment by DavBfr** [comment]
Fixed in Printing 3.7.1.
Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).
